### PR TITLE
Run common and dashboard tests also with newest available allowed packages for all deps

### PR DESCRIPTION
**Pull request by sh-rp**
### Description
We have tests to test lower bounds of our allowed dependencies. We also want to see if a newer version of any or our dependencies (as allowed in pyproject.toml) breaks anything in dlt.

### CODE REVIEW

File: tests/common/schema/test_coercion.py [hunk 1]
```diff
@@ -32,9 +32,10 @@ def test_coerce_type_to_text() -> None:
     # bytes to text (base64)
     assert coerce_value("text", "binary", b"binary string") == "YmluYXJ5IHN0cmluZw=="
     # HexBytes to text (hex with prefix)
-    assert (
-        coerce_value("text", "binary", HexBytes(b"binary string")) == "0x62696e61727920737472696e67"
-    )
+    assert coerce_value("text", "binary", HexBytes(b"binary string")) in [
```
Comment (sh-rp) [inline tests/common/schema/test_coercion.py:35]: A newer version of hexbytes returns the hexcode without "0x". We have a couple of options:
* Keep as is, this means users will see other values in their destination after upgrading this lib
* Normalize result in normalizer so the result is the same regardless of lib version (kind of a breaking change, either for users on the old version or the new version)
* Update min version of hexbytes to the version with the new output. Also a kind of breaking change.

Imho this is probably an edge case. I think I would probably keep it as is.

Comment (rudolfix) [reply tests/common/schema/test_coercion.py:35]: HexBytes was used to make sure that 0x is present... so if that got removed it does not make sense to use it. My take: write our own HexBytes that alwasy prefixes and parses the prefix. it should be pretty simple to do.

for now I'd merge this ticket

**Review by rudolfix — APPROVED** [review]
@sh-rp let's merge it. I always wanted to test of max dependencies too.

re. HexBytes: see my comment I bet @alkaline-0 can port this to dlt and remove the dependency :)
